tests/testthat/test-derive_var_dthcaus.R

```diff
@@ -437,3 +437,115 @@ test_that("dthcaus_source Test 9: returns a warning when traceability_vars is us
     class = "lifecycle_warning_deprecated"
   )
 })
+
+## Test 10: multiple observations from different sources ----
+test_that("derive_var_dthcaus Test 10: multiple observations from different sources", {
+  expected <- tibble::tribble(
+    ~STUDYID, ~USUBJID, ~DTHCAUS,
+    "TEST01", "PAT01",  "SUDDEN DEATH",
+    "TEST01", "PAT02",  NA_character_,
+    "TEST01", "PAT03",  "DEATH DUE TO progression of disease"
+  )
+
+  adsl <- select(expected, -DTHCAUS)
+
+  ae <- tibble::tribble(
+    ~STUDYID, ~USUBJID, ~AESEQ, ~AEDECOD, ~AEOUT, ~AEDTHDTC,
+    "TEST01", "PAT01", 12, "SUDDEN DEATH", "FATAL", "2021-04-04"
+  ) %>%
+    mutate(
+      AEDTHDT = ymd(AEDTHDTC)
+    )
+
+  ds <- tibble::tribble(
+    ~STUDYID, ~USUBJID, ~DSSEQ, ~DSDECOD, ~DSTERM, ~DSSTDTC,
+    "TEST01", "PAT01", 4, "DEATH", "DEATH DUE TO progression of disease", "2021-04-05",
+    "TEST01", "PAT02", 1, "INFORMED CONSENT OBTAINED", "INFORMED CONSENT OBTAINED", "2021-04-02",
+    "TEST01", "PAT02", 2, "RANDOMIZATION", "RANDOMIZATION", "2021-04-11",
+    "TEST01", "PAT02", 3, "COMPLETED", "PROTOCOL COMPLETED", "2021-12-01",
+    "TEST01", "PAT03", 1, "DEATH", "DEATH DUE TO progression of disease", "2021-04-07",
+    "TEST01", "PAT03", 2, "RANDOMIZATION", "RANDOMIZATION", "2021-04-11",
+    "TEST01", "PAT03", 3, "COMPLETED", "PROTOCOL COMPLETED", "2021-12-01"
+  )
```

Comment:
Do we need to worry about any partial dates here since these are `---DTC` variables?

Reply:
I guess a death would always have an exact date.  just musing here

Reply:
No need to worry. `derive_var_dthcaus()` expects dates (not `DTC` variables). The users have to handle partial dates in a preprocessing step or on the definition of the source objects (like in the examples).